lib/views/reporting/widgets/manager/location.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:geolocator/geolocator.dart';
+import 'package:latlong2/latlong.dart';
+import 'package:permission_handler/permission_handler.dart';
+import 'package:flutter/material.dart';
+import 'package:wilde_buren/views/home/home_view.dart';
+
+class LocationManager {
+  Future<LatLng> getUserLocation(BuildContext context) async {
+    final status = await _getLocationPermission();
+    if (status.isGranted) {
+      Position position = await Geolocator.getCurrentPosition(
+        desiredAccuracy: LocationAccuracy.high,
+      );
+      return LatLng(position.latitude, position.longitude);
+    } else if (status.isDenied) {
+      if (!context.mounted) {
+        return const LatLng(51.25851739912562, 5.622422796819703);
+      }
+      requestLocationAccess(context);
+    } else {
+      if (!context.mounted) {
+        return const LatLng(51.25851739912562, 5.622422796819703);
+      }
+      _showLocationPermissionDialog(context);
+    }
+    return const LatLng(51.25851739912562, 5.622422796819703);
+  }
+
+  Future<PermissionStatus> _getLocationPermission() async {
+    return await Permission.location.request();
+  }
+
+  Future<void> requestLocationAccess(BuildContext context) async {
+    await _getLocationPermission();
+    return;
```

Comment:
Deze return kan in principe weggelaten worden maar het kan geen kwaad dat hij er staat